flake-module.nix

```diff
@@ -192,6 +188,13 @@ ihpFlake:
 
                 env.IHP_LIB = "${ihp}/lib/IHP";
                 env.IHP = "${ihp}/lib/IHP"; # Used in the Makefile
+
+                scripts.deploy-to-nixos.exec = ''
+                    echo "usage: deploy-to-nixos <flake> <target-host>"
+                    echo "example: deploy-to-nixos stagging.example.com"
+                    nix shell nixpkgs#nixos-rebuild --command bash -c "nixos-rebuild switch -j auto --use-substitutes --fast --flake .#$1 --target-host $1 --build-host $1 --option substituters https://digitallyinduced.cachix.org --option trusted-public-keys digitallyinduced.cachix.org:digitallyinduced.cachix.org-1:y+wQvrnxQ+PdEsCt91rmvv39qRCYzEgGQaldK26hCKE="
+                    # ssh ihp-deploy-test systemctl start migrate
+                '';
             };
```

Comment:
What if for example I need to run some custom custom build steps, for example a custom make command, or npm that some people use? Would it work with what's being done locally on the machine?

Reply:
Yes that should work by adding custom build steps to the `Makefile`. E.g. by making your custom build task a dependency of `build/bin/RunUnoptimizedProdServer`. The it will be triggered automatically during the `nix build` call